mmv1/templates/terraform/custom_flatten/pubsub_no_wrapper_write_metadata_flatten.go.erb

```diff
@@ -0,0 +1,28 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2023 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+	if v == nil {
+		return nil
+	}
+
+	original := v.(map[string]interface{})
+	transformed := make(map[string]interface{})
+
+	if original["writeMetadata"] == nil {
+		transformed["write_metadata"] = false
+	}
```

Comment:
```suggestion
	if original["writeMetadata"] == nil {
		transformed["write_metadata"] = false
	} else {
		transformed["write_metadata"] =  original["writeMetadata"]
	}
```

Looks like the value isn't getting set now in the case of true. This should fix it.